tests/TestUtilities.py

```diff
@@ -13,16 +13,18 @@ def execute_coala(func, binary, *args, stdout_only=False):
     :param function:    A main function from coala_json, coala_ci module etc.
     :param binary:      A binary to execute coala test
     :param stdout_only: Get the stdout output only, discard stderr
-    :return:            A tuple holding a return value first and
-                        a stdout output as second element.
+    :return:            A tuple holding a return value as first element,
+                        a stdout output as second element and a stderr output
+                        as third element if stdout_only is False.
     """
     sys.argv = [binary] + list(args)
     with retrieve_stdout() as stdout:
         with retrieve_stderr() as stderr:
             retval = func()
-            return (retval,
-                    stdout.getvalue() if stdout_only else
-                    stdout.getvalue() + '\n' + stderr.getvalue())
+            output = (retval, stdout.getvalue())
+            if not stdout_only:
```

Comment:
if stdout_only is true, we should fetch stderr and ensure it is empty.  raise an exception with the unexpected content in the exception message.